geomesa-process-utils/src/main/scala/org.locationtech.geomesa.process.utils/KryoLazyDensityUtils.scala

```diff
@@ -0,0 +1,82 @@
+/***********************************************************************
+* Copyright (c) 2013-2016 Commonwealth Computer Research, Inc.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Apache License, Version 2.0
+* which accompanies this distribution and is available at
+* http://www.opensource.org/licenses/apache2.0.php.
+*************************************************************************/
+
+package org.locationtech.geomesa.process.utils
+
+import com.vividsolutions.jts.geom.Envelope
+import org.locationtech.geomesa.features.ScalaSimpleFeature
+import org.locationtech.geomesa.features.SerializationOption.SerializationOptions
+import org.locationtech.geomesa.features.kryo.KryoFeatureSerializer
+import org.locationtech.geomesa.utils.geotools.{GeometryUtils, GridSnap}
+import org.locationtech.geomesa.utils.interop.SimpleFeatureTypes
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable
+
+object KryoLazyDensityUtils {
```

Comment:
let's put this in `geomesa-index-api/src/main/scala/org/locationtech/geomesa/index/iterators` instead of creating a whole new module